File: src/com/javascript-practice/linkList/createLinkedList.js
class Node {
    constructor(val) {
      this.val = val;
      this.next = null;
    }
  }

const createLinkedList = (values, i = 0) => {
    if( i === values.length) return null;
    const head = new Node(values[i]);
    head.next = createLinkedList(values, i + 1);
    return head;
}

let head = createLinkedList(["h", "e", "y"]);
// h -> e -> y
console.log(head);
while(head!=null) {
    console.log(head.val);
    head  = head.next;
}

File: src/com/structy/LinkedList/CreateLinkedList.java
package com.structy.LinkedList;

public class CreateLinkedList {

    public Node createLinkedList(int[]nums){
        Node dummy = new Node(-1);
        Node tail = dummy;
        for (int num:nums) {
            tail.next = new Node(num);
            tail = tail.next;
        }
        return dummy.next;
    }
    public Node createLinkedListRec(int[]nums){
        return helper(nums, 0);
    }
    private Node helper(int[]nums, int start){
        if(nums.length == start){
            return null;
        }
        Node head = new Node(nums[start]);
        head.next = helper(nums, start+1);
        return head;
    }

    public static void main(String[] args) {
        CreateLinkedList createLinkedList = new CreateLinkedList();
        int[]nums = {1, 7, 1, 8};
        Node result = createLinkedList.createLinkedListRec(nums);
        Node tmp = result;
        while (tmp!=null){
            System.out.print(" ->" + tmp.val);
            tmp = tmp.next;
        }
    }
}
